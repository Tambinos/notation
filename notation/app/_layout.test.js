import React from "react";
import renderer from "react-test-renderer";
import Layout from "../app/_layout"; // adjust path if needed

// Mock external dependencies
jest.mock("expo-router", () => ({
    router: { push: jest.fn() },
    Slot: () => <></>,
}));

jest.mock("expo-document-picker", () => ({
    getDocumentAsync: jest.fn(),
}));

jest.mock("react-native/Libraries/Alert/Alert", () => ({
    alert: jest.fn(),
}));

jest.mock("../utils/AsyncStorage", () => ({
    setItem: jest.fn(),
}));

describe("Layout Component", () => {
    it("renders correctly and matches the snapshot", () => {
        const tree = renderer.create(<Layout />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
