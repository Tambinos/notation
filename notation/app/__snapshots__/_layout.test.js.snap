// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component renders correctly and matches the snapshot 1`] = `null`;
